service/storage_service.cc

```diff
@@ -1128,6 +1130,21 @@ class topology_coordinator {
                 co_await update_topology_state(take_guard(std::move(node)), {builder.build()}, "rebuilding completed");
             }
                 break;
+            case node_state::left_token_ring: {
+                // Wait until other nodes observe the new token ring and stop sending writes to this node.
+                bool exec_command_res;
+                std::tie(node, exec_command_res) = co_await exec_global_command(
+                        std::move(node), raft_topology_cmd::command::barrier, false);
```

Comment:
We need to wait for all writes to complete, no?

Reply:
Yes.

So we need another command like `fence_old_writes`?

Reply:
We need to rebase on @gusev-p fencing series that already have the drain verb.

Reply:
Or leave a FIXME and merge in a follow-up :P